synapse/handlers/federation.py

```diff
@@ -427,13 +436,33 @@ async def try_backfill(domains: Collection[str]) -> bool:
                     continue
                 except HttpResponseException as e:
                     if 400 <= e.code < 500:
-                        raise e.to_synapse_error()
+                        logger.warning(
+                            "Backfill denied from %s because %s [%d/%d]",
+                            dom,
+                            e,
+                            denied_count,
+                            max_denied_count,
+                        )
+                        denied_count += 1
+                        if denied_count >= max_denied_count:
+                            return False
```

Comment:
`return False` and not backfilling makes the most sense to me 👍

vs throwing a `4xx` or `5xx`

This topic was mentioned in [backend internal](https://matrix.to/#/!SGNQGPGUwtcPBUotTL:matrix.org/$R_6Biur-3is7HoAtYMA4w9JSb7iIg5pJm2L0yRdpTmA?via=jki.re&via=matrix.org&via=element.io).

---

Ideally for a client expecting to see messages in their `/messages` response we could indicate that there is a gap ([MSC3871](https://github.com/matrix-org/matrix-spec-proposals/pull/3871))